scripts/build/Jenkins/Jenkinsfile

```diff
@@ -220,7 +268,35 @@ def GetCrashArtifactDir() {
     }
 }
 
-def CheckoutBootstrapScripts(String branchName) {
+def GetRemoteConfig(repositoryName, repositoryUrl) {
+    def theRemoteConfig = [
+        name: repositoryName,
+        url: repositoryUrl,
+        credentialsId: scm.userRemoteConfigs.credentialsId[0]
```

Comment:
As @brianherrera noted below, the `userRemoteConfigs` object has a few more methods than what's defined here. [See here for details](https://javadoc.jenkins.io/plugin/git/hudson/plugins/git/UserRemoteConfig.html). Maybe pulling in the object in full can help define some of the missing methods.

Alternatively, we can add a wider range refspec method for pull requests, like

```
refspec:'+refs/pull/*/head:refs/remotes/origin/*'
```

Reply:
Discussed it with @byrcolin. It looks like the `scm.userRemoteConfigs` object is expressing some methods that are needed for it to pull in changes, but only for the engine checkout. We do want to investigate why this occurs, but for now, the suggested changes look like these commits: 
https://github.com/aws-lumberyard-dev/o3de/commit/36f32b5e8277b0fffdf14c87f5045572f80f2ac8
https://github.com/aws-lumberyard-dev/o3de/commit/eb2616ce2bf24ad785a900f181663350a3baf684

Reply:
Merged in these suggestions and AR has built successfully